src/Umbraco.Infrastructure/Persistence/Repositories/Implement/TrackedReferencesRepository.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NPoco;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Persistence;
+using Umbraco.Cms.Core.Persistence.Repositories;
+using Umbraco.Cms.Core.Scoping;
+using Umbraco.Cms.Infrastructure.Persistence.Dtos;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement
+{
+    internal class TrackedReferencesRepository : ITrackedReferencesRepository
+    {
+        private readonly IScopeAccessor _scopeAccessor;
+
+        public TrackedReferencesRepository(IScopeAccessor scopeAccessor)
+        {
+            _scopeAccessor = scopeAccessor;
+        }
+
+        public IEnumerable<RelationItem> GetPagedItemsWithRelations(int[] ids, long pageIndex, int pageSize,
```

Comment:
It would be good to have an overload that takes just on id. This saves on creating the array when this method is called from the Controller.

Reply:
In v9.4.0rc, GetPagedRelationForItems is being called each time a node is navigated to in the backoffice (might be due to having listviews). Unfortunately this is throwing SQL timeout exception, resulting in an error being shown to the editors. Needs some further optimization before this would be safe to use.

For the overload, the current code path is already being used with one nodeid, having an overload avoids the need to allocate the array for what appears to be a hot path for those using the backoffice.

Reply:
Yes, `GetPagedRelationsForItem` is being called quite often but it is different from `GetPagedItemsWithRelations` - [link](https://github.com/umbraco/Umbraco-CMS/pull/12146/files#diff-b093bee8d7ba0778d115425d915af6df526ae4cd727d8aced91141ef12344046R18) and it has changed its signature as we expect to pass 1 item at a time ;) so the issue should be fixed now

Reply:
I also added a fix, so the request is not made before the info tab is shown